src/pages/Quiz/Quiz.tsx

```diff
@@ -0,0 +1,146 @@
+import { useContext, useEffect, useState } from "react";
+import { useParams } from "react-router-dom";
+import Data from "../../assets/stages.json";
+import { AuthContext } from "../../context/auth.context";
+import { getRandomElement, shuffle } from "../../helpers/question";
+import { storeNewScoreService } from "../../services/score.service";
+import { Check } from "./check";
+import { QuestionContainer } from "./questionContainer";
+
+export type STATES = {
+  totalPoint: number;
+  changeTotalPoint: () => void;
+  questionNum: number;
```

Comment:
I would recommend you to refactor this. 

May be good to start considering separating things by context

eg. questionNum is for navigation (if I understood correctly)
eg. totalPoint seems like game specific (if I understood correctly)

and consider bulding custom hooks and contexts. (Please do not put everyhitng into one hook or context hehe)